tests/templates/test_repeat.py

```diff
@@ -0,0 +1,131 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the :func:`pennylane.template.repeat` function.
+Integration tests should be placed into ``test_templates.py``.
+"""
+# pylint: disable=protected-access,cell-var-from-loop
+import pytest
+import pennylane as qml
+from pennylane import repeat, template
+
+@template
+def ConstantCircuit():
+
+    qml.PauliX(wires=[0])
+    qml.Hadamard(wires=[0])
+    qml.PauliY(wires=[1])
+
+@template
+def StaticCircuit(wires, var):
+
+    qml.CNOT(wires=[wires[3], wires[1]])
+    qml.Hadamard(wires=wires[1])
+    qml.PauliY(wires=wires[2])
+
+    if var == True:
+        qml.Hadamard(wires=wires[0])
+
+@template
+def KwargCircuit(wires, **kwargs):
+
+    qml.CNOT(wires=[wires[3], wires[1]])
+    qml.Hadamard(wires=wires[1])
+    qml.PauliY(wires=wires[2])
+
+    if kwargs['var'] == True:
+        qml.Hadamard(wires=wires[0])
+
+@template
+def DynamicCircuit(parameters):
+
+    for i in range(2):
+        qml.RX(parameters[0][i], wires=i)
+
+    qml.MultiRZ(parameters[1], wires=[0, 1])
+
+@template
+def MultiCircuit(parameters1, parameters2, var1, wires, var2):
+
+    if var2 == True:
+        for i, w in enumerate(wires):
+            qml.RY(parameters1[i], wires=w)
+
+    if var1 == True:
+        qml.templates.BasicEntanglerLayers([parameters2], wires=wires)
+
+UNITARIES = [
+    ConstantCircuit,
+    StaticCircuit,
+    KwargCircuit,
+    DynamicCircuit,
+    MultiCircuit
+]
+
+########################
+
+
+class TestRepeat:
+    """Tests the repetition function"""
+
+    def test_depth_error(self):
+        """Tests that the correct error is thrown when depth is not an integer"""
+
+        depth = 1.5
+
+        def unitary(wires):
+            qml.PauliX(wires=wires)
+
+        with pytest.raises(ValueError, match=r"'depth' must be of type int"):
+            repeat(unitary, depth, wires=[0])
+
+    def test_args_length(self):
+        """Tests that the correct error is thrown when the length of an argument is incorrect"""
+
+        params = [1, 1]
+
+        def unitary(param, wire):
+            qml.RX(param, wires=wire)
+
+        with pytest.raises(ValueError, match=r"Each argument in args must have length matching 'depth'; expected 3"):
+            repeat(unitary, 3, params, wires=[0])
+
```

Comment:
I would put test data on top of the file? Although in this case it is easy to read!